src/scatter.jl

```diff
@@ -0,0 +1,40 @@
+for op in [+, -, *, /, max, min, &, |]
+    @eval function scatter_kernel!(op::typeof($(op)), dst, src, idx)
```

Comment:
```suggestion
    function scatter_kernel!(op, dst, src, idx)
```
why don't we define the kernel for generic ops and let it error out by itself for CUDA unsupported ones?

Reply:
Done.